src/main/java/pl/edu/agh/firecell/core/statebuilder/element/IgnitionElement.java

```diff
@@ -0,0 +1,22 @@
+package pl.edu.agh.firecell.core.statebuilder.element;
+
+import org.joml.Vector3i;
+import pl.edu.agh.firecell.model.Cell;
+import pl.edu.agh.firecell.model.util.IndexUtils;
+
+import java.util.List;
+
+public record IgnitionElement(Vector3i position, Vector3i size) implements BaseCuboidElement {
+
+    @Override
+    public void addTo(List<Cell> cells, Vector3i spaceSize) {
+        IndexUtils.range(position, new Vector3i(-1).add(position).add(size))
+                .map(expandedIndex -> IndexUtils.flattenIndex(expandedIndex, spaceSize))
+                .filter(index -> cells.get(index).flammable())
+                .forEach(index -> {
+                    var oldCell = cells.get(index);
+                    cells.set(index, new Cell(400, 1, oldCell.flammable(),
```

Comment:
Here you set an arbitrary start temperature. That's wrong, take it from the material of the cell.